pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -447,10 +442,12 @@ public void publishMessage(ByteBuf headersAndPayload, PublishContext publishCont
 
     public void updateSubscribeRateLimiter() {
         SubscribeRate subscribeRate = this.getSubscribeRate();
-        if (isSubscribeRateEnabled(subscribeRate)) {
-            subscribeRateLimiter = Optional.of(subscribeRateLimiter.orElse(new SubscribeRateLimiter(this)));
-        } else {
-            subscribeRateLimiter = Optional.empty();
+        synchronized (subscribeRateLimiter) {
```

Comment:
`synchronized(subscribeRateLimiter)` seems wrong. I can now see that this came from the original code. This should be fixed.
One possible fix is to make the `subscribeRateLimiter` field a `volatile` field.

Reply:
Thanks a lot. I have updated it. PTAL @lhotari @Jason918